### PR TITLE
Code Quality: Use WindowContext instead of AppModel

### PR DESCRIPTION
**Pull request by 0x5bfa**
AppModel was leftover from UWP, removed and replaced with WindowContext which can be DI'd.

### CODE REVIEW

File: src/Files.App/Data/Contexts/Window/IWindowContext.cs
Comment (yaira2) [inline src/Files.App/Data/Contexts/Window/IWindowContext.cs:1]: Can you add docs before we merge?

**Comment by yaira2** [comment]
For testing, we should test one or two of the properties and ensure they are updated correctly. For example, we should ensure that `IsCompactOverlay` and `IsPasteEnabled` are correctly updated when the state changes.

File: src/Files.App/Data/Contexts/Window/IWindowContext.cs
Comment (0x5bfa) [reply src/Files.App/Data/Contexts/Window/IWindowContext.cs:1]: Sure.

File: src/Files.App/Data/Contexts/Window/WindowContext.cs [hunk 1]
```diff
@@ -1,27 +1,143 @@
...
-			SetProperty(
-				ref isCompactOverlay,
-				e.Kind is AppWindowPresenterKind.CompactOverlay,
-				nameof(IsCompactOverlay));
+			IsCompactOverlay = e.Kind is AppWindowPresenterKind.CompactOverlay;
+		}
+
+		private void Clipboard_ContentChanged(object? sender, object e)
+		{
+			// TODO: Refactor here
```
[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.App/Data/Contexts/Window/WindowContext.cs:130]: Idk. I'll find using GitHub Blame.

[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.App/Data/Contexts/Window/WindowContext.cs:130]: It looks like it was you: https://github.com/files-community/Files/commit/b495c18a0617b1cbf52d416bc14d9f4e91ceaf41#diff-3d8909aba7bd61ebd4096358aff8c4ba62c9382ce2caff5636391476d0845116R25

I don’t feel there needs of a refactoring here. Rather, it’s the best place to do so.

File: src/Files.App/Data/Contexts/Window/IWindowContext.cs
Comment (0x5bfa) [reply src/Files.App/Data/Contexts/Window/IWindowContext.cs:1]: Done.

File: src/Files.App/Data/Contexts/Window/WindowContext.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (yaira2) [reply src/Files.App/Data/Contexts/Window/WindowContext.cs:138]: I don't remember why but it's possible I wanted to move that method out of AppModel.

File: src/Files.App/Actions/Navigation/NextTabAction.cs [hunk 2]
```diff
@@ -28,7 +28,7 @@ public NextTabAction()
 
 		public Task ExecuteAsync()
 		{
-			App.WindowContext.TabStripSelectedIndex = (App.WindowContext.TabStripSelectedIndex + 1) % multitaskingContext.TabCount;
+			App.WindowContext.SelectedTabBarItemIndex = (App.WindowContext.SelectedTabBarItemIndex + 1) % multitaskingContext.TabCount;
```
Comment (yaira2) [inline src/Files.App/Actions/Navigation/NextTabAction.cs:31]: `Selected` should be after `TabBar`.

File: src/Files.App/Data/Contexts/Window/WindowContext.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (yaira2) [reply src/Files.App/Data/Contexts/Window/WindowContext.cs:138]: Sounds good, let's remove the comment then.

[the remark this answers is not in the document]
Comment (0x5bfa) [reply src/Files.App/Data/Contexts/Window/WindowContext.cs:138]: Done.

**Comment by 0x5bfa** [comment]
Now, ready to be merged

**Review by yaira2 — APPROVED** [review]

**Comment by 0x5bfa** [comment]
I'm not sure why it's fail again and again since I can build, deploy and see that the window successfully opens without any crash.

It's failing here:
https://github.com/files-community/Files/blob/cc0a7701c9c2b744136a8f58940463dd769978e1/tests/Files.InteractionTests/Tests/FolderTests.cs#L54

**Comment by yaira2** [comment]
Try rebasing the branch from `main`.

**Comment by yaira2** [comment]
I rebased this PR from `main` so that should solve the CI issues.

**Comment by yaira2** [comment]
@0x5bfa can you try forking these changes to another branch to see if that fixes the CI issues?

**Comment by 0x5bfa** [comment]
Sure.

**Pull request closed without merging**